lstchain/io/event_selection.py

```diff
@@ -65,12 +78,101 @@ class DL3FixedCuts(Component):
     ).tag(config=True)
 
     def gh_cut(self, data):
+        """
+        Applying a fixed global gammaness cut on a given data
+        """
         return data[data["gh_score"] > self.fixed_gh_cut]
 
+    def optimize_gh_cuts(
+        self, data, energy_bins, min_value=0.1,
+        max_value=0.99, smoothing=None, min_events=10
+    ):
+        """
+        Evaluating an optimized energy-dependent gammaness cuts, in a given
+        data, with provided reco energy bins, and other parameters to
+        pass to the pyirf.cuts.calculate_percentile_cut function
+        """
+        #To be sure to have the efficiency in between 0 and 100
+        if self.fixed_gh_efficiency < 1:
+            self.fixed_gh_efficiency *= 100
+
+        gh_cuts = calculate_percentile_cut(
+            data["gh_score"],
+            data["reco_energy"],
+            bins = energy_bins,
+            min_value = min_value,
+            max_value = max_value,
+            fill_value = data["gh_score"].max(),
+            percentile = 100 - self.fixed_gh_efficiency,
+            smoothing = smoothing,
+            min_events = min_events,
+        )
+        return gh_cuts
+
+    def apply_optimized_gh_cuts(self, data, gh_cuts):
```

Comment:
`energy_dependent` instead of `optimized`. A single cut can also be `optimized`

Reply:
Ok, will amend it.